src/state/opamSwitchState.ml

```diff
@@ -392,11 +398,12 @@ let load lock_kind gt rt switch =
     else switch_config
   in
   let switch_config, switch_invariant =
-    if OpamVersion.compare
-        switch_config.OpamFile.Switch_config.opam_version
-        (OpamVersion.of_string "2.1")
-       >= 0
-    || switch_config.OpamFile.Switch_config.invariant <> OpamFormula.Empty
+  (* tweak this check, but with what ? *)
```

Comment:
At the moment, `format_version` in `OpamFile.Switch_config` is no longer used - was the plan still to include that in the switch config?

I don't think we _need_ to (the `opam-root-version` in the root config file is enough), but wouldn't `format_version` be the `opam-config-version` field proposed [in the wiki](https://github.com/ocaml/opam/wiki/opam-root-compatibilities#fields-to-set-up)?

Reply:
We don't need to have them in the file, we can always retrieve the information with config's `opam-root-version`. But internally it is still useful to have a quick lookup and a better following of the version updates.